### PR TITLE
Update Q6

### PR DESCRIPTION
**Pull request by adophoxia**
## Description
Superseeds https://github.com/qmk/qmk_firmware/pull/21423, but this will only cover one board from each series update of PRs. Also enables Num Lock Indicator LED.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/keychron/q6/ansi/info.json [hunk 1]
```diff
@@ -36,7 +143,7 @@
                 {"matrix": [1, 19], "x": 19.5, "y": 0},
                 {"matrix": [2, 19], "x": 20.5, "y": 0},
                 {"matrix": [3, 19], "x": 21.5, "y": 0},
-
+
```
Comment (fauxpark) [inline keyboards/keychron/q6/ansi/info.json:146]:
```suggestion

```

File: keyboards/keychron/q6/ansi/info.json [hunk 2]
```diff
@@ -114,7 +221,7 @@
                 {"matrix": [4, 17], "x": 18.5, "y": 4.25},
                 {"matrix": [4, 18], "x": 19.5, "y": 4.25},
                 {"matrix": [4, 14], "x": 20.5, "y": 4.25},
-
+
```
Comment (fauxpark) [inline keyboards/keychron/q6/ansi/info.json:224]:
```suggestion

```

File: keyboards/keychron/q6/ansi_encoder/info.json [hunk 3]
```diff
@@ -120,7 +228,7 @@
                 {"matrix": [4, 17], "x": 18.5, "y": 4.25},
                 {"matrix": [4, 18], "x": 19.5, "y": 4.25},
                 {"matrix": [4, 14], "x": 20.5, "y": 4.25},
-
+
```
Comment (fauxpark) [inline keyboards/keychron/q6/ansi_encoder/info.json:231]:
```suggestion

```

File: keyboards/keychron/q6/iso/info.json [hunk 4]
```diff
@@ -58,7 +166,7 @@
                 {"matrix": [1, 18], "x": 19.5, "y": 1.25},
                 {"matrix": [0, 17], "x": 20.5, "y": 1.25},
                 {"matrix": [0, 18], "x": 21.5, "y": 1.25},
-
+
```
Comment (fauxpark) [inline keyboards/keychron/q6/iso/info.json:169]:
```suggestion

```

File: keyboards/keychron/q6/iso/info.json [hunk 5]
```diff
@@ -115,7 +223,7 @@
                 {"matrix": [4, 17], "x": 18.5, "y": 4.25},
                 {"matrix": [4, 18], "x": 19.5, "y": 4.25},
                 {"matrix": [4, 14], "x": 20.5, "y": 4.25},
-
+
```
Comment (fauxpark) [inline keyboards/keychron/q6/iso/info.json:226]:
```suggestion

```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Aerocatia** [comment]
Is it alright if you don't enable the numlock light by default? A few of us find that [annoying](https://old.reddit.com/r/Keychron/comments/12vhug2/i_regret_upgrading_my_firmware_to_v15_q6/).

**Comment by fauxpark** [comment]
The default num lock state on powerup is controlled by the BIOS. These changes simply assign a LED in the RGB matrix as the num lock indicator.

**Comment by Aerocatia** [comment]
Yeah, I am aware of this. The lock state LED indicator on these boards is what some find obnoxious, as it is shared with the key LED. I assume that is why it was commented out in the first place as assigning it in the RGB matrix never did not work, but I could be wrong.

**Comment by fauxpark** [comment]
I'm not really seeing what the issue is...just turn it off in the BIOS?

**Comment by Aerocatia** [comment]
The issue is not the numlock state itself, it's the desire to have the key LED unlit when numlock is engaged because it looks better this way. This is the current behavior and this PR changes it. 

Anyway, this is not hard to change locally before compiling the firmware, so I do not mind if it changes that much.

**Comment by fauxpark** [comment]
I guess it could be made a little more configurable - I was assuming the LED would be left unchanged if num lock was off but it is indeed set to 0,0,0. And the on/off colours are essentially hardcoded in `q6.c`.

**Pull request closed without merging**